### DISCUSSION TITLE
how to hash the password via CLI

### DISCUSSION THREAD

**Discussion by msimerson** [the-djmaze/snappymail#1133] [Q&A] [answered]
When I install snappymail, I have access to the password I'd like to set for the admin user. In looking at the hash value for admin_password, I noticed it's bcrypt and tried hashing it with PHP like so;

````
echo 'secret pass' | php -r 'echo password_hash(stream_get_contents(STDIN), PASSWORD_DEFAULT);'
$2y$10$88/3QGCMyC754oUA4LcwYewkLHp/Artf6kY52Clycw8QpjyvES0Oe
````

That does produce a bcrypt hash but when I store that hash in application.ini, I can't use it to log in. How do I hash a password so that I can store the hash in application.ini and snappymail will be happy?

**Comment by the-djmaze** [comment]
Why not just use the admin interface that?

**Reply by msimerson** [reply]
I'm automating the install process and I want the installer to keep working in the future. Establishing a login session with a web client and POSTing a form to update the password is fragile because the backend form is subject to change. Is there a stable API for setting that password via the admin interface?

**Reply by the-djmaze** [reply, marked as answer]
```php
<?php
$_ENV['SNAPPYMAIL_INCLUDE_AS_API'] = true;
require_once '/var/lib/snappymail/index.php';

$oConfig = \RainLoop\Api::Config();
$oConfig->SetPassword('mypassword input from somewhere');
```
